orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py

```diff
@@ -454,10 +562,84 @@ def testToyBertCheckpointLoadZero():
     actual_eval_loss = trainer.eval_step(input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels)
     actual_eval_loss = actual_eval_loss.cpu().numpy().item(0)
 
+    checkpoint_files = sorted(checkpoint._list_checkpoint_files(ckpt_dir, 'bert_toy_lamb'))
+    loaded_state_dict = checkpoint.experimental_state_dict(trainer)
+    loaded_state_dict = checkpoint._split_state_dict(loaded_state_dict)
+
+    for f in checkpoint_files:
+        rank_state_dict = torch.load(f, map_location=torch.device("cpu"))
+        rank_state_dict = checkpoint._split_state_dict(rank_state_dict)
+
+        for k,v in rank_state_dict['fp16_param'].items():
+            fp32_name = k.split('_fp16')[0]
+            assert_allclose(v, loaded_state_dict['fp32_param'][fp32_name])
+            assert_allclose(v, loaded_state_dict['fp16_param'][k])
+        
+        for k,v in rank_state_dict['optimizer'].items():
+            if k in loaded_state_dict['optimizer']:
+                assert_allclose(v, loaded_state_dict['optimizer'][k])
+
     # Check results
     assert_allclose(expected_eval_loss, actual_eval_loss, rtol=rtol)
 
 
+# checkpointfiles generated using:
+# mpirun -n 4 python3 orttrainer_bert_toy_onnx_ckpt_gen.py
+def testToyBertCheckpointLoadZeroMixedprecisionAdam():
```

Comment:
Same as the full precision case. If the difference from the previous test is the optimizer, please parameterize it and keep a single implementation